src/Resource/CloudAccount/Endpoint.php

```diff
@@ -148,6 +152,165 @@ protected function _waitUntilPhpVersion(
     };
   }
 
+  /**
+   * Create a backup
+   *
+   * @param Entity An instance of cloud account entity.
+   * @return Backup
+   * @throws ApiException If request fails
+   */
+  public function createBackup(Entity $entity) : Backup {
+    $this->_wait(null);
+    $response = $this->_client->request(
+      'POST',
+      self::_URI . "/{$entity->getId()}/backup"
+    );
+
+    return $this->getModel(Backup::class)->sync($response);
+  }
+
+  /**
+   * Return a list of backups
+   *
+   * @return Backup
+   * @throws ApiException If request fails
+   */
+  public function getBackups(Entity $entity) : Collection {
+    $this->_wait(null);
+    $collection = new Collection(Backup::class);
+
+    foreach ($this->_fetchBackupList($entity) as $backup) {
+      $collection->add($this->getModel(Backup::class)->sync($backup));
+    }
+
+    return $collection;
+  }
+
+  /**
+   * Return a specific backup
+   *
+   * @param string $file_name The unique file name for the backup to retrieve.
+   * @return Backup
+   * @throws ApiException If request fails
+   */
+  public function getBackup(Entity $entity, string $file_name) : Backup {
+    $this->_wait(null);
+    return $this->_findBackup($entity, $file_name);
+  }
+
+  /**
+   * Download a specific backup
+   *
+   * @param string $file_name The unique file name for the backup to retrieve.
+   * @param string $path the directory to store the download in.
+   *
+   * @return Promise
+   * @throws ApiException If request fails
+   * @throws Exception
+   */
+  public function downloadBackup(
+    Entity $entity,
+    string $file_name,
+    string $path
+  ) {
+    $this->_wait(null);
+
+    if (! file_exists($path) || ! is_dir($path)) {
+      throw new CloudAccountException(
+        CloudAccountException::INVALID_PATH,
+        ['filepath' => $path]
+      );
+    }
+
+    $path = trim($path);
+    if (substr($path, -1) !== DIRECTORY_SEPARATOR) {
+      $path .= DIRECTORY_SEPARATOR;
+    }
+
+    $save_to = $path . $file_name;
+    if (file_exists($save_to)) {
+      throw new CloudAccountException(
+        CloudAccountException::FILE_EXISTS,
+        ['filename' => $save_to]
+      );
+    }
+
+    $stream = @fopen($save_to, 'w');
+    
+    if (! is_resource($stream)) {
+      throw new CloudAccountException(
+        CloudAccountException::INVALID_STREAM,
+        ['filename' => $save_to]
+      );
+    }
+
+    try {
+      $this->_client->request(
+        'GET',
+        $this->_findBackup($entity, $file_name)->get('download_url'),
+        [
+          'cookies' => (new CookieJar()),
+          'sink' => $stream,
+          'verify' => false
+        ]
+      );
+    } catch (\Exception $e) {
+      fclose($stream);
+      unlink($save_to);
+      throw new CloudAccountException($e);
```

Comment:
do we expect any specific exception here, aside from `BACKUP_NOT_FOUND`?

since we have nothing useful to add to the exception message/context, and especially since `BACKUP_NOT_FOUND` is _already_ a `CloudAccountException`, we should just rethrow it as-is.
```php
fclose($stream);
unlink($save_to);
throw $e;
```